great_expectations/gx/plugins/expectations/expect_column_values_to_have_list_length.py

```diff
@@ -0,0 +1,142 @@
+import pandas as pd
+from typing import Optional, Any
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.execution_engine import PandasExecutionEngine
+from great_expectations.expectations.expectation import ColumnMapExpectation
+from great_expectations.expectations.metrics import (
+    ColumnMapMetricProvider,
+    column_condition_partial,
+)
+
+
+# This class defines a Metric to support your Expectation.
+# For most ColumnMapExpectations, the main business logic for calculation will live in this class.
+class ColumnValuesListLength(ColumnMapMetricProvider):
+    """Class definition for list length checking metric."""
+
+    # This is the id string that will be used to reference your metric.
+    condition_metric_name = "column_values.list_length"
+    condition_value_keys = ("list_length",)
+
+    # This method implements the core logic for the PandasExecutionEngine
+    @column_condition_partial(engine=PandasExecutionEngine)
+    def _pandas(cls, column: pd.core.series.Series, list_length: int, **kwargs) -> bool:
+        """Core logic for list length checking metric on a
+        pandas execution engine.
+
+        Args:
+            column (pd.core.series.Series): Pandas column to be evaluated.
```

Comment:
What are your opinions removing typing from the docstring? We already explicitly have the typing in the function definition.

Reply:
The custom expectations are created from a [template](https://github.com/great-expectations/great_expectations/blob/develop/examples/expectations/column_map_expectation_template.py). The docstring is part of the template and is a standard way that expectations contributed to `great_expectations` are created. Because we may want to contribute these custom expectations to the open-source project in the future, I think it's best if they stay formatted similarly to the template.

Reply:
Sorry, I was looking at the wrong line when I posted my original comment. I agree that it is a bit redundant, but other code in this project has type hints and typing in the docstring (though it is not 100% consistent)